### PR TITLE
Untrack DialogReference when dialog is closed

### LINKED ISSUE #2450: Shortcut dialog stops working after clicking close
The shortcut dialog stops working after clicking close.

Repo:
1. Click (?) button in top right. Dialog opens.
2. Click x or Close in the dialog.
3. Click (?) button in top right. Nothing happens.

shortcut-dialog

### PR DESCRIPTION
**Pull request by tlmii**
Fixes #2450.

The `IDialogReference` instance created when either the settings or help dialogs were shown was never cleared when they were closed. 
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2465)

### CODE REVIEW

**Review by JamesNK — APPROVED** [review]
LGTM but don't merge yet. The shortcut PR might be reverted - although perhaps on only P4 branch. PRs on top of it will make that more complicated.

Feel free to fix things but wait a day before merging PRs for clarity on what's happening here.

cc @kvenkatrajan @adamint

**Review by adamint — APPROVED** [review]

**Pull request merged**
